tests/functional/SaveQueryValidateCest.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+// "query FooQuery { post( id: "1", idType: DATABASE_ID) { title } }"   <-- this is the one we want to error on !!!!!
+// "query ($num: ID!) { post( id: "1", idType: DATABASE_ID) { title } }"
+// "query ($num: ID!) { post( id: $num, idType: DATABASE_ID) { title } }"
+//        looking for  [4] => argument: (id) ()\n            [5] => variable: (num)\n
```

Comment:
Oh, my notes are in here.  Ha!

Reply:
We'd want to error on all 3 of these examples. 

The problem we're trying to prevent, is folks doing string concatenation for their queries and instead push folks toward using variables. 

All 3 of those queries could be concatenated like so: 

---
example: `query FooQuery { post( id: "1", idType: DATABASE_ID) { title } }`

```js
const id = 123;
const query = `query FooQuery { post( id: ${id}, idType: DATABASE_ID) { title } }`
```

---

example: `query ($num: ID!) { post( id: "1", idType: DATABASE_ID) { title } }` (perhaps this was a mistake, but this looks like it might be invalid already as an argument is defined  but not used)

```js
const id = 123;
const idType: "DATABASE_ID";
const query = `query ($num: ID!) { post( id: ${id}, idType: ${idType}) { title } }`
```

---

example: `query ($num: ID!) { post( id: $num, idType: DATABASE_ID) { title } }`

```js
const idType = "DATABASE_ID";
const query = `query ($num: ID!) { post( id: $num, idType: ${idType}) { title } }`
```

----

In each of these cases we have arguments passed in to fields that do not map to an argument defined for the query. 

The expected "valid" definition of the query would be: 

```graphql
query ($id: ID! $idType: PostIdType ) {
  post( id: $id idType: $idType ) {
    title
  }
}
```

Here, the field has 2 arguments, and _both_ are defined in the operation. 

If either of the arguments are not defined on the operation and used at the field level as an argument, then we might still be letting concatenated query strings persist many permutations of the same query, which is the situation we're trying to prevent.

----